core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -166,10 +193,17 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
     const minFontSize = 50;
     const maxFontSize = 150;
 
-    const d3Scale = d3.scaleLinear();
-    // const d3Scale = d3.scaleSqrt();
-    // const d3Scale = d3.scaleLog();
-
+    const getScale: () => d3.ScaleContinuousNumeric<number, number> = () => {
+      switch (this.wordCloudControls.scale) {
+        case 'linear':
+          return d3.scaleLinear();
+        case 'log':
+          return d3.scaleLog();
+        case 'square root':
+          return d3.scaleSqrt();
+      }
+    };
```

Comment:
My personal preference is to define these short functions as inline functions. The main purpose to use a function here rather than just assign the variable using if/else is to let typescript does exhaustive type checking